### PR TITLE
GP11 : Time functions fix/panic reason

### PR DESCRIPTION
**Pull request by cedric-chaumont-st-dev**
Signed-off-by: Cedric Chaumont <cedric.chaumont@st.com>

### CODE REVIEW

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

File: core/tee/tee_svc.c [hunk 1]
```diff
@@ -897,12 +897,15 @@ TEE_Result tee_svc_get_time(enum utee_time_category cat, TEE_Time *mytime)
...
-	if (res == TEE_SUCCESS || res == TEE_ERROR_OVERFLOW) {
-		res2 = tee_svc_copy_to_user(s, mytime, &t, sizeof(t));
-		if (res2 != TEE_SUCCESS)
-			res = res2;
+	if (res != TEE_SUCCESS && res != TEE_ERROR_OVERFLOW) {
+		t.seconds = 0;
+		t.millis = 0;
 	}
 
+	res2 = tee_svc_copy_to_user(s, mytime, &t, sizeof(t));
```
Comment (jenswi-linaro) [inline core/tee/tee_svc.c]: Now we're always copying back the time even if we didn't get anything useful, why?

Comment (cedric-chaumont-st-dev) [inline core/tee/tee_svc.c]: If an error other than TEE_ERROR_OVERFLOW is returned, this structure is filled with zeroes.

Comment (jenswi-linaro) [inline core/tee/tee_svc.c]: How is that useful?

**Comment by cedric-chaumont-st-dev** [comment]
Updated.

**Comment by cedric-chaumont-st-dev** [comment]
Updated.

File: lib/libutee/tee_api.c [hunk 2]
```diff
@@ -202,12 +202,37 @@ TEE_Result TEE_Wait(uint32_t timeout)
 
 TEE_Result TEE_GetTAPersistentTime(TEE_Time *time)
 {
-	return utee_get_time(UTEE_TIME_CAT_TA_PERSISTENT, time);
+	TEE_Result res;
+
+	res = utee_get_time(UTEE_TIME_CAT_TA_PERSISTENT, time);
+
+	if (res != TEE_SUCCESS && res != TEE_ERROR_OVERFLOW) {
+		time->seconds = 0;
```
Comment (jenswi-linaro) [inline lib/libutee/tee_api.c]: So now this has moved into this function instead. I still don't understand why we should set it to zero though. The application shouldn't have any expectations on what's in *time if the function doesn't return `TEE_SUCCESS` or `TEE_ERROR_OVERFLOW`.

Comment (cedric-chaumont-st-dev) [inline lib/libutee/tee_api.c]: @jenswi-linaro: "If an error other than TEE_ERROR_OVERFLOW is returned, this structure is filled with zeroes." What is your understanding about this ?

Comment (jenswi-linaro) [inline lib/libutee/tee_api.c]: Sorry, I missed that. Yes this is the way to do it.

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by cedric-chaumont-st-dev** [comment]
Rebased.Squashed.Tested (ATM, ARM Juno).
28701 subtests of which 0 failed
687 test cases of which 0 failed
0 test case was skipped
TEE test application done!

**Pull request merged**